thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/Entity.java

```diff
@@ -31,4 +41,31 @@ public String getUrn() {
   public double getScore() {
     return score;
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    Entity entity = (Entity) o;
+
+    if (Double.compare(entity.score, score) != 0) {
+      return false;
```

Comment:
Try using Objects.hash and Objects.equals, it will save many lines of code and improve readability and consistency with rest of the code